### PR TITLE
from TimeSampledCorrection to TimeIntervalCorrection

### PR DESCRIPTION
**Pull request by GiovanniVolta**
Given the fact that for average SEG we are going to use one value per science run, I have changed the mother class of `AvgSingleElectronGain` from `TimeSampledCorrection` to `TimeIntervalCorrection`.

Currently, on Mondo DB, we already have an avg_se_gain v1 version, which was implemented for TimeSampledCorrection. Will be a bit in conflict with this PR? 

Maybe to avoid any problem we could release a new version of xedocs?

### CODE REVIEW

**Comment by shenyangshi** [comment]
Hi @GiovanniVolta, sounds good to me. The other thing is that the example notebook is changed, you can do git checkout path_to_file origin master to get it back :)

**Comment by GiovanniVolta** [comment]
Ops! Thank you, I did not notice it

**Comment by flammhead** [comment]
Since, I have no idea, what I am doing, I tried to play a bit around...
I have checked out the above branch and installed the local xedocs.
Then I tried initializing the modified class to check if anything crashes. I get the following output

`(XENONnT_development) [fjoerg@midway3-login3 software]$ python
Python 3.9.19 (main, Mar 21 2024, 17:11:28) 
[GCC 11.2.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import xedocs
>>> test_instance = xedocs.schemas.corrections.avg_se_gain.AvgSingleElectronGain()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pydantic/main.py", line 341, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 3 validation errors for AvgSingleElectronGain
version
  field required (type=value_error.missing)
time
  field required (type=value_error.missing)
value
  field required (type=value_error.missing)`

To be honest, I don't even know if this is good or bad, neither if my approach would be the correct one to verify that the PR is good to go.

**Review by flammhead — DISMISSED** [review]
The changes look reasonable to me and are in line with what we discussed in the meeting. Also the notebook was changed back to the original state, so I would say, this PR is good to go. I assume my check in initializing the class was not the right approach :D.

**Review by LuisSanchez25 — COMMENTED** [review]
You dont ned to import the TimeSampledCorrections here since you arent using it

**Comment by Ananthu-Ravindran** [comment]
Do we still need the TimeSampledCorrections?

**Comment by LuisSanchez25** [comment]
> Since, I have no idea, what I am doing, I tried to play a bit around... I have checked out the above branch and installed the local xedocs. Then I tried initializing the modified class to check if anything crashes. I get the following output
> 
> ```
> 
> (XENONnT_development) [fjoerg@midway3-login3 software]$ python
> Python 3.9.19 (main, Mar 21 2024, 17:11:28) 
> [GCC 11.2.0] :: Anaconda, Inc. on linux
> Type "help", "copyright", "credits" or "license" for more information.
> >>> import xedocs
> >>> test_instance = xedocs.schemas.corrections.avg_se_gain.AvgSingleElectronGain()
> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
>   File "pydantic/main.py", line 341, in pydantic.main.BaseModel.__init__
> pydantic.error_wrappers.ValidationError: 3 validation errors for AvgSingleElectronGain
> version
>   field required (type=value_error.missing)
> time
>   field required (type=value_error.missing)
> value
>   field required (type=value_error.missing)
> ```
> 
> To be honest, I don't even know if this is good or bad, neither if my approach would be the correct one to verify that the PR is good to go.

This is working as intended, you tried to create an instance of the class without giving it the requiered fields

File: xedocs/schemas/corrections/implementations/avg_se_gain.py [hunk 1]
```diff
@@ -9,11 +9,11 @@
 
 import rframe
 
-from ..base_corrections import TimeSampledCorrection
+from ..base_corrections import TimeSampledCorrection, TimeIntervalCorrection
```
Comment (flammhead) [inline xedocs/schemas/corrections/implementations/avg_se_gain.py:12]:
```suggestion
from ..base_corrections import TimeIntervalCorrection
```

**Review by flammhead — APPROVED** [review]

**Review by Ananthu-Ravindran — APPROVED** [review]
Hi @GiovanniVolta , this looks good!

**Comment by LuisSanchez25** [comment]
Ok this all looks good now, there are currently versions for ONLINE and v1, so when adding avg_se_gains corrections they should by v2 and onwards.
Everything looks good to me, I think this is ready to be merged if everyone else is happy.

**Pull request merged**